R/mlm-utils.R

```diff
@@ -0,0 +1,65 @@
+#' create_mlm_matrix
+#' @noRd
+create_mlm_matrix <- function(dl, value){
+  lapply(dl, \(z) as.matrix(z[value]))
+}
+
+#' get_mlm_values
+#' @noRd
+get_mlm_values <- function(res, n_units){
+  malmquist <- c(rep(NA_real_, n_units), do.call('c', lapply(res, \(x) x$malmquist))) |> as.vector()
+  effch <- c(rep(NA_real_, n_units), do.call('c', lapply(res, \(x) x$effch))) |> as.vector()
+  tech <- c(rep(NA_real_, n_units), do.call('c', lapply(res, \(x) x$tech))) |> as.vector()
+  obtech <- c(rep(NA_real_, n_units), do.call('c', lapply(res, \(x) x$obtech))) |> as.vector()
+  ibtech <- c(rep(NA_real_, n_units), do.call('c', lapply(res, \(x) x$ibtech))) |> as.vector()
+  matech <- c(rep(NA_real_, n_units), do.call('c', lapply(res, \(x) x$matech))) |> as.vector()
+  scale_effch <- c(rep(NA_real_, n_units), do.call('c', lapply(res, \(x) x$scale_effch))) |> as.vector()
+  pure_effch <- c(rep(NA_real_, n_units), do.call('c', lapply(res, \(x) x$pure_effch))) |> as.vector()
```

Comment:
I suggest changing all of these statements swapping `do.call()` with `vapply()` as it is far more efficient and safer:
```r
identical(c(rep(NA_real_, n_units), vapply(res, \(x) x$malmquist, numeric(n_units))) |> as.vector()
```